src/map-server.ts

```diff
@@ -99,6 +99,19 @@ export default class MapServer {
     return this.#api.listTilesets(baseApiUrl)
   }
 
+  /**
+   * Create a tileset. Returns the created tileset TileJSON if successful.
+   */
+  createTileset(
+    tileset: Readonly<TileJSON>,
```

Comment:
Making this `Readonly` required a few small tweaks to downstream functions, such as `getTilesetId`. I prefer adding `Readonly` because (1) I think it's clearer (2) it prevents a class of bug, but I don't feel strongly.

Reply:
was a little worried that it require casting the argument to `ReadOnly` when calling the function, but that doesn't seem to be the case so I'm for it 👍